### DISCUSSION TITLE
Change RGB byte order

### DISCUSSION THREAD

**Discussion by andiasergio** [hyperion-project/hyperion.ng#1555] [Q&A]
Hi,

It's possible change to GRB byte order from command line?

I want to use in a headless systems, then I need configure the correct byte order for my leds strip with an script

Thanks.

**Comment by Lord-Grey** [comment]
Not sure, if I really get the problem you would like to solve…
Even if you run Hyperion on a headless server, you configure the LED-Device via the Web-UI which includes the byte order.
After you defined the LED Device you can use the ByteOrder Wizard to change is as required.
Or you switch the expert level settings mode and set the byteorder while configuring the LEDDevice.

Why do you need to switch the byte order after you configured the LED-Device?

**Reply by andiasergio** [reply]
I'll like to preconfigure the LED-device, I don't want give others access to the web-ui, because of that I'll like to create a configuration script for automatic installation, but I don't know how set the byte order from command line.

**Comment by Lord-Grey** [comment]
Ok. Got it.
There is currently no way exposed to change the configuration persistently from outside.

Of course you still have the option to update the color order in the Hyperion configuration DB via MySQL.

**Comment by Lord-Grey** [comment]
Looks like there is a configSet option offered by hyperion-remote
Maybe you check, if that is of help.

You might first need do a configGet, change the respective value and do a set afterwards.
I will have a look how the JSON for a „device“ needs to look like.

**Reply by andiasergio** [reply]
I'll try with that, thanks.

**Reply by andiasergio** [reply]
Hi,
I tried with:
`hyperion-remote --configGet`
`hyperion-remote -a 192.168.1.30 --configGet`
`hyperion-remote -a 192.168.1.30:19444 --configGet`

Always have this response:
`2023-01-03T11:18:47.721 |__| REMOTE            : <ERROR> No configuration file available in result`

Any idea what is wrong?

**Reply by Lord-Grey** [reply]
Hi @andiasergio 

> Any idea what is wrong?

Looks like the way the reply is constructed was changed over time. Therefore the expected keyword is not found to extract the response.
I have done a fix, see PR #1549. 
If the build is finished you can download a package and test the fixed hyperion-remote.

Then you could do a 

1. `hyperion-remote --configGet > myConfig.txt`
2. Remove all output before the config JSON-object 
3. Update the ColorOrder
4. `hyperion-remote --configSet "$(cat myConfig.txt)"`

It looks like you always need to set a complete configuration.

**Reply by andiasergio** [reply]
I get some errors when I try to compile with this command:
`cmake -DCMAKE_BUILD_TYPE=Release ..`

Then, when I try use **make** I get this error:
`make: *** No targets specified and no makefile found.  Stop.`

**Reply by Lord-Grey** [reply]
Why do you not just download a ready made build from the PR
https://github.com/hyperion-project/hyperion.ng/actions/runs/3831042315
Look at the artifacts.

**Reply by Lord-Grey** [reply]
> I get some errors when I try to compile with this command:

If you have cmake errors, then you should resolve them before running make. I would assume that you did not install all the dependent libraries?

**Reply by andiasergio** [reply]
Hi again, 
I get config information with --configGet, but I can't set with --configSet always get error this error:
`<ERROR> Failed to parse json data from hyperion-remote-args: Error: illegal value at Line: 0, Column: 1, Data: 'cfg2.json'`
I tryed with the exact response o get, I tryed remove and left all of json info, I tryed with stringify, I tryed with and without Configuration attribute, nothing works.
Can you give me a valid config json file?

**Comment by Lord-Grey** [comment]
> Can you give me a valid config json file?

Here you go:
[Hyperion-2.0.15-beta.1-Backup.json.txt](https://github.com/hyperion-project/hyperion.ng/files/10349978/Hyperion-2.0.15-beta.1-Backup.json.txt)

You can also do an Export from Hyperion-UI and get a complete file per instance.
Note: If you have multiple instances, you unfortunately need to do a getConfig/setConfic per instance.
